src/components/templates/AddStory.js

```diff
@@ -23,13 +23,24 @@ const AddStory = ({ toggleModal }) => {
       !newStory.title.trim()
       || !newStory.content.trim()
     ) return;
-
+    //don't attempt if the user is not logged in
+    if (
+      !currentUser
+    ) return;
+    //don't attempt if there is no local user date
+    if (
+      !Date.now()
+    ) return;
     try {
       await db.collection('links')
         .add({
           title: newStory.title,
           content: newStory.content,
           author: currentUser.displayName,
```

Comment:
Let's change author to authorName to be a little more explicit now that we have two author fields in one document
```suggestion
          authorName: currentUser.displayName,
```